splunk_add_on_ucc_framework/install_python_libraries.py

```diff
@@ -69,38 +101,41 @@ def install_python_libraries(
     path_to_requirements_file = os.path.join(source_path, "lib", "requirements.txt")
     if os.path.isfile(path_to_requirements_file):
         logger.info(f"Installing requirements from {path_to_requirements_file}")
-        if includes_ui:
-            ucc_library_present = _check_ucc_library_in_requirements_file(
-                path_to_requirements_file
-            )
-            if not ucc_library_present:
-                raise SplunktaucclibNotFound(
-                    f"splunktaucclib is not found in {path_to_requirements_file}. "
-                    f"Please add it there because this add-on has UI."
-                )
         if not os.path.exists(ucc_lib_target):
             os.makedirs(ucc_lib_target)
         install_libraries(
-            path_to_requirements_file,
-            ucc_lib_target,
-            python_binary_name,
+            requirements_file_path=path_to_requirements_file,
+            installation_path=ucc_lib_target,
+            installer=python_binary_name,
         )
-
-        if os_libraries:
-            install_os_dependent_libraries(
-                ucc_lib_target, python_binary_name, os_libraries
+        if includes_ui and not _pip_is_lib_installed(
+            installer=python_binary_name,
+            target=ucc_lib_target,
+            libname="splunktaucclib",
+        ):
+            raise SplunktaucclibNotFound(
+                f"splunktaucclib is not found in {path_to_requirements_file}. "
+                f"Please add it there because this add-on has UI."
             )
 
-        packages_to_remove = (
+        cleanup_libraries = install_os_dependent_libraries(
+            ucc_lib_target=ucc_lib_target,
+            installer=python_binary_name,
+            os_libraries=os_libraries,
+        )
+
+        packages_to_remove = {
             "setuptools",
             "bin",
             "pip",
             "distribute",
             "wheel",
-        )
-        remove_package_from_installed_path(
-            ucc_lib_target,
-            packages_to_remove,
+        }
+        # we can remove os-dependent libraries from the installation_path to save some space.
+        packages_to_remove.update(cleanup_libraries)
```

Comment:
Just to confirm - it will remove `cryptography` from `package/lib` when `cryptography` is present in OS-dependent configuration?

Reply:
correct